vispy/visuals/colorbar.py

```diff
@@ -0,0 +1,330 @@
+            self._program = ModularProgram(VERT_SHADER, FRAG_SHADER_VERTICAL)
+        else:
+            raise ColorBarVisual._get_orientation_error(self._orientation)
+
+        tex_coords = np.array([[0, 0], [1, 0], [1, 1],
+                              [0, 0], [1, 1], [0, 1]],
+                              dtype=np.float32)
+
+        self._program.frag['color_transform'] = Function(self._cmap.glsl_map)
+        self._program['a_texcoord'] = tex_coords.astype(np.float32)
+
+        self.regenerate()
+
+    def regenerate(self):
+        x, y = self._center_pos
+        halfw, halfh = self._halfdim
+
+        if self._orientation == "horizontal":
+            text_x, text_y = x, y + halfh * 1.2 + self.border_width
+
+            self._label.pos = text_x, text_y
+
+            begin_tick_pos = x - halfw, text_y
+            end_tick_pos = x + halfw, text_y
+
+            # TODO, HACK: This should ideally be a single TextVisual
+            # However, one TextVisual with multiple strings
+            # does not seem to be working as of now. (See #981)
+            # https://github.com/vispy/vispy/issues/981
+            self._ticks[0].pos = begin_tick_pos
+            self._ticks[1].pos = end_tick_pos
+
+            self._ticks[0].anchor_y = self.ticks[1].anchor_y = "top"
+
+        elif self._orientation == "vertical":
+            text_x, text_y = x + halfw * 1.2 + self._border_width, y
+
+            self._label.pos = text_x, text_y
+            self._label.rotation = -90
+            self._label.anchor_y = "top"
+
+            begin_tick_pos = text_x, y + halfh
+            end_tick_pos = text_x, y - halfh
+
+            # TODO, HACK: See comment about ticks on "horizontal" conditional
+            self._ticks[0].pos = begin_tick_pos
+            self._ticks[1].pos = end_tick_pos
+
+            self._ticks[0].anchor_y = self.ticks[1].anchor_y = "top"
+            self._ticks[0].rotation = self.ticks[1].rotation = -90
+
+        else:
+            raise ColorBarVisual._get_orientation_error(self._orientation)
+
+        vertices = np.array([[x - halfw, y - halfh],
+                            [x + halfw, y - halfh],
+                            [x + halfw, y + halfh],
+                             # tri 2
+                            [x - halfw, y - halfh],
+                            [x + halfw, y + halfh],
+                            [x - halfw, y + halfh]],
+                            dtype=np.float32)
+
+        self._program['a_position'] = vertices.astype(np.float32)
+        self._border_program['a_position'] = vertices.astype(np.float32)
+
+        self._border_program.vert['border_width'] = self._border_width
+        self._border_program.frag['border_color'] = self._border_color.rgba
+
+    @staticmethod
+    def _get_orientation_error(orientation):
+        return ValueError("orientation must"
+                          " be \'horizontal\' or \'vertical\'."
+                          " given: %s" % orientation)
```

Comment:
I would say `... or 'vertical', not '%s'."` Also, the single quotes do not need to be escaped. Finally, it is safer to encapsulate `orientation` in a tuple as `"..." % (orientation, )`.